### ISSUE TITLE
Idea: the refiner's gate is ~13% of every refinement step — does a blend coefficient need 2d x d?

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#211]
**Lane:** partial-cpu — the toy-scale ablation runs on CPU via `experiments/depth/ablation_harness.py`; a confirming LM-scale pair wants the card.
**Type:** `ideas` — it changes what the model is.

## The cost nobody has priced

`trm/model/refiner.py:166` gates every refinement step with a full projection over the concatenated pair:

```python
self.gate = nnx.Linear(2 * dim, dim, ...)
...
g = jax.nn.sigmoid(self.gate(jnp.concatenate([z_new, z], axis=-1)))
return g * z_new + (1.0 - g) * z
```

At dim 960 that is `2 × 960 × 960 = 1.84M MAC/position`, paid **once per refinement iteration**. Against the block's ~12.04M, the gate is **~13% of every refine step** — and unlike the block, it is pure control machinery: it produces no representation, only a mixing coefficient.

At depth 8 the gate alone is ~14.7M MAC/position, roughly the cost of adding a **1.2th encoder block** to every forward.

## The question

Does a mixing coefficient need `2d × d` of capacity? Cheaper parameterizations, in increasing order of how much they give up:

1. **Per-channel gate** — `g` from a `Linear(2*dim, dim)` replaced by a learned per-channel vector modulated by a low-rank read of `[z_new, z]`. Rank `r`: `2·d·r + r·d` MAC — at r=64 that is ~15% of current.
2. **Scalar gate per position** — `Linear(2*dim, 1)`, one blend weight for the whole vector. ~0.05% of current cost. This is the aggressive end.
3. **Fixed schedule** — no learned gate at all; `g` a function of step index only (the `gate_bias` init already encodes a prior that early steps should be retention-biased).

Option 3 is nearly free and is the sharpest test of whether the gate is doing anything data-dependent, or just implementing a depth-varying step size.

## Why this is not settled by the literature

Highway networks and Universal Transformers use full gates, but neither is answering "how much gate capacity does depth recurrence need" — they are answering "does gating help at all." The `gate_bias` finding here (balanced steps compound and diverge at depth 8; retention-bias fixes it — `ablation_results.md` run 2) suggests the gate's *useful* role may be closer to a learned step-size schedule than to a rich per-channel decision. That is testable and, as far as I can tell, untested.

## Pre-registration (fill in before running)

- **Harness:** `experiments/depth/ablation_harness.py` — toy task at the exact shipped arch, cheap enough to run all four arms.
- **Arms:** full gate (control) / low-rank r=64 / scalar / fixed schedule. **Same seed, same data order.**
- **Kill/keep:** an arm survives if it stays within 2σ of the full gate on the toy metric, where σ comes from a seed sweep of the control **measured first**. Survivors get a matched LM-scale pair; anything that loses at toy scale is dead without burning GPU time.
- Expect this to fail at the scalar end. A **negative** here is worth recording either way — "the refiner's gate genuinely needs per-channel capacity" is a novel finding about depth recurrence and belongs in `docs/findings/`.

## Blocked by

Nothing for the toy arm (CPU, runs any time). The LM-scale confirmation is behind the base run (#157) and the base-model bar.

**Comment by MatthewLacerda2** [comment]
superseded-by the depth-recurrence kill (2026-09-12): this concerns the refine loop, which is no longer the live architecture. Record: docs/findings/2026-09-12-depth-recurrence-is-suppressed-not-exploited.md and the ROADMAP graveyard. Re-file against a live mechanism if it ever applies to one.
